### PR TITLE
CLOUD: Add a load button to fetch JSON from a file

### PR DESCRIPTION
**Pull request by lephilousophe**
This feature has been requested on Discord.

This can be used on handhelds where there is no browser.

The user does the login process on its desktop, saves the JSON in a file and uploads it to its handheld through an unspecified mean and loads it on ScummVM using this new button.

### CODE REVIEW

File: gui/cloudconnectionwizard.cpp [hunk 1]
```diff
@@ -355,6 +359,7 @@ void CloudConnectionWizard::showStepManualMode2() {
 	_label0 = new StaticTextWidget(_container, "ConnectionWizard_ManualModeStep2.Line1", _("Copy the JSON code from the browser here and press Next:"));
 	_codeBox = new EditTextWidget(_container, "ConnectionWizard_ManualModeStep2.CodeBox", Common::U32String(), Common::U32String(), 0, 0, ThemeEngine::kFontStyleConsole);
 	_button0 = new ButtonWidget(_container, "ConnectionWizard_ManualModeStep2.PasteButton", _("Paste"), _("Paste code from clipboard"), kCloudConnectionWizardPasteCodeCmd);
+	_button1 = new ButtonWidget(_container, "ConnectionWizard_ManualModeStep2.LoadButton", _("Load"), _("Load code from storage"), kCloudConnectionWizardLoadCodeCmd);
```
Comment (sev-) [inline gui/cloudconnectionwizard.cpp:362]: "Load code from file" ?

File: gui/cloudconnectionwizard.cpp [hunk 2]
```diff
@@ -620,6 +626,22 @@ void CloudConnectionWizard::handleCommand(CommandSender *sender, uint32 cmd, uin
 		}
 		break;
 
+	case kCloudConnectionWizardLoadCodeCmd: {
+		BrowserDialog browser(_("Select JSON file from ScummVM"), false);
```
Comment (sev-) [inline gui/cloudconnectionwizard.cpp:630]: // I18N: JSON is a file format name
"Select JSON file copied from scummvm.org site"

File: gui/cloudconnectionwizard.cpp [hunk 3]
```diff
@@ -620,6 +626,22 @@ void CloudConnectionWizard::handleCommand(CommandSender *sender, uint32 cmd, uin
...
+		if (browser.runModal() > 0) {
+			// User made his choice...
+			Common::File codeFile;
+			if (!codeFile.open(browser.getResult())) {
+				MessageDialog alert(_("Failed to load JSON file"));
```
Comment (sev-) [inline gui/cloudconnectionwizard.cpp:635]: // I18N: JSON is a file format name

**Review by sev- — APPROVED** [review]
Reviewed, couple small suggestions related to the wording

File: gui/cloudconnectionwizard.cpp (hunk 2 above)
Comment (lephilousophe) [reply gui/cloudconnectionwizard.cpp]: Thanks. I don't know where my mind was when I wrote that. :smile:

**Comment by lephilousophe** [comment]
Thanks for the review.
I will merge later today.

**Comment by sev-** [comment]
Thanks!

**Pull request merged**